pkg/health/server.go

```diff
@@ -33,16 +34,53 @@ type Server interface {
 }
 
 type server struct {
-	ready *atomic.Bool
-	log   logger.Logger
+	ready  *atomic.Bool
+	router http.Handler
+	log    logger.Logger
+}
+
+type RouterOptions func() (string, http.Handler)
+
+func NewRouterOptions(path string, handler http.Handler) RouterOptions {
+	return func() (string, http.Handler) {
+		return path, handler
+	}
+}
+
+func NewJsonDataRouterOptions[T any](path string, getter func() (T, error)) RouterOptions {
+	return NewRouterOptions(path, http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
```

Comment:
IMHO the /healthz endpoint should not have any data in it, because it's meant to just respond with a 2xx status code indicating everything is ok. It is invoked multiple times per minute on K8s, so it needs to be as fast as possible.

This is especially important in this case, as placement tables can have a lot of data, so this endpoint is very slow (plus, it allocates a lot of memory due to copying the table).

Can this be implemented as a separate endpoint rather than /healthz? For example `GET /placement/state` or anything else.

Reply:
> IMHO the /healthz endpoint should not have any data in it, because it's meant to just respond with a 2xx status code indicating everything is ok. It is invoked multiple times per minute on K8s, so it needs to be as fast as possible.

Yes, this totally makes sense! And in fact, it is indeed implemented as a separate endpoint, the variable `path` in this function is the endpoint for the handler. It just uses the same port as health check.
The endpoint currently is set to `/tables`, as it is included in the placement server, I haven't add a `/placement` prefix.

Reply:
Ohh, I missed that it was a separate endpoint.

Ok, makes sense. However I would like one of @dapr/maintainers-dapr to confirm what the API endpoint should be. I think a prefix is needed.

Reply:
I am ok with `/placement/state`, changed,  use this endpoint unless there are further ideas on that. Tips, I am also ok with something like `/placement/actor` or `/actor/state`.